pkg/ddb/expr_filter.go

```diff
@@ -28,3 +30,32 @@ func (b *filterBuilder) buildFilterCondition(metadata *Metadata) *expression.Con
 
 	return &expr
 }
+
+type ttlStruct struct {
+	Ttl int64 `json:"ttl"`
+}
+
+type ttlFilterer interface {
+	PerformFilterCondition(metadata *Metadata, item map[string]*dynamodb.AttributeValue, clock clock.Clock) (bool, error)
+}
+
+func (b *filterBuilder) PerformFilterCondition(metadata *Metadata, item map[string]*dynamodb.AttributeValue, clock clock.Clock) (bool, error) {
```

Comment:
Wouldn't it be nicer, leaner and cleaner if you would take metadata and clock from the builder instead of providing it as parameters?